### PR TITLE
Text slug

### PR DESCRIPTION
**Pull request by jansta**
Implements #343

### CODE REVIEW

**Comment by jansta** [comment]
Not sure, if this is what was requested though ;)

File: src/main/java/apoc/text/Strings.java [hunk 1]
```diff
@@ -160,4 +160,12 @@ public String format(@Name("text") String text, @Name("params") List<Object> par
         if (params == null) return text;
         return String.format(Locale.ENGLISH,text, params.toArray());
     }
+    
+    @UserFunction
+    @Description("apoc.text.slug(text, delim) - slug the text with the given delimiter")
+    public String slug(@Name("text") String text, @Name(value = "delim", defaultValue = "-") String delim) {
+        if (text == null) return null;
+        if (delim == null) return null;
+        return text.replace(" ", delim).replace(delim+delim, delim);
```
Comment (jexp) [inline src/main/java/apoc/text/Strings.java]: probably use replaceAll("\s+",delim) for all whitespace
or \W+ for non-word-characters

**Comment by jexp** [comment]
Looks good, please have a look at my comment. Perhpas also test for subsequent space?

**Comment by jansta** [comment]
Better like this? :) Anything else to improve?

**Comment by jexp** [comment]
something is wrong now, there is a lot of other changes, can you rebase your work onto the latest master state? and then push to the PR branch again

**Comment by jansta** [comment]
sure, sorry :)

File: src/main/java/apoc/text/Strings.java [hunk 2]
```diff
@@ -160,4 +160,15 @@ public String format(@Name("text") String text, @Name("params") List<Object> par
         if (params == null) return text;
         return String.format(Locale.ENGLISH,text, params.toArray());
     }
+    
+    @UserFunction
+    @Description("apoc.text.slug(text, delim) - slug the text with the given delimiter")
+    public String slug(@Name("text") String text, @Name(value = "delim", defaultValue = "-") String delim) {
+        if (text == null) return null;
+        if (delim == null) return null;
+        return text.trim()
```
Comment (jexp) [inline src/main/java/apoc/text/Strings.java]: it's actually much easier  `.replaceAll("[\\W\\s]+","delim")

**Comment by jexp** [comment]
Thanks a lot !!

**Comment by tomswinkels** [comment]
I found one problem, the slug must be always set to lowercase.

**Pull request merged**